### PR TITLE
Bail early for files which don't match the requested extensions

### PR DESCRIPTION
**Pull request by ocean90**
We shouldn't calculate match scores for files which will be ignored.

### CODE REVIEW

File: src/IterableCodeExtractor.php [hunk 1]
```diff
@@ -223,7 +227,7 @@ function ( $file, $key, $iterator ) use ( $include, $exclude, $extensions ) {
 						return static::containsMatchingChildren( $file, $include );
 					}
 
-					return ( ( $inclusion_score >= $exclusion_score ) && $file->isFile() && in_array( $file->getExtension(), $extensions, true ) );
+					return $inclusion_score >= $exclusion_score;
```
Comment (schlessera) [inline src/IterableCodeExtractor.php]: This has now changed the logic, as we would now return a score for non-Files as well.
```suggestion
					return $inclusion_score >= $exclusion_score && $file->isFile();
```
@swissspidy Am I missing something here?

Comment (swissspidy) [reply src/IterableCodeExtractor.php]: Nope, you're right.

Could we perhaps change the condition above to:

```php
if ( ! $file->isFile() || ( $file->isFile() && ! in_array( $file->getExtension(), $extensions, true ) ) ) {
	return false;
}
```

instead?

Comment (schlessera) [reply src/IterableCodeExtractor.php]: The second `$file->isFile()` is redundant. If the first one succeeds, it stops there, if it fails, then `isFile()` is guaranteed to be true.

Comment (schlessera) [reply src/IterableCodeExtractor.php]: So:
```php
if ( ! $file->isFile() || ! in_array( $file->getExtension(), $extensions, true ) ) {
	return false;
}
```

**Comment by schlessera** [comment]
This change seems to cause a lot of breakage.

**Pull request closed without merging**